src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -1348,6 +1361,65 @@ void primitive_inst::do_runtime_in_place_concat() {
     GPU_DEBUG_TRACE_DETAIL << "[In place concat] " << concat_inst->id() << ": can_be_optimized " << std::endl;
 }
 
+void primitive_inst::do_runtime_in_place_crop() {
+    OV_ITT_SCOPED_TASK(ov::intel_gpu::itt::domains::intel_gpu_plugin, openvino::itt::handle("do_runtime_in_place_crop: " + id()));
+    GPU_DEBUG_GET_INSTANCE(debug_config);
+    GPU_DEBUG_IF(debug_config->disable_runtime_buffer_fusing) {
+        return;
+    }
+
+    for (auto u : get_user_insts()) {
+        if (u->get_node().is_type<crop>()) {
+            if (u->get_node().can_be_optimized()) {
+                GPU_DEBUG_TRACE_DETAIL << "[In place crop] update shape for " << u->id() << std::endl;
+                u->update_shape();
+                u->update_shape_done_by_other = true;
+
+                const auto& crop_users = u->get_user_insts();
+                std::vector<layout> reshape_layouts;
+                if (crop_users.size() == 1 && crop_users.front()->get_node().is_type<reshape>()) {
```

Comment:
Is "users.size() > 1" case handled somewhere? (maybe I missed)

Reply:
added exception handling related to user size as discussed